### PR TITLE
CDC Postgres Tests

### PR DESCRIPTION
**Pull request by cgardens**
## What
* Adds tests. Fixes issues along the way.

## How
*Describe the solution*

## Pre-merge Checklist
- [ ] *Figure out breaking behavior for deletions*
- [ ] *Fix helper to assert one state message*
- [ ] *Add test that check AirbyteRecordMessage metadata fields.*

## Recommended reading order
1. `PostgresSourceCdcTest.java`
1. `DebeziumRecordPublisherTest.java`
1. the rest

### CODE REVIEW

File: airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/DebeziumRecordPublisher.java [hunk 1]
```diff
@@ -121,7 +123,7 @@ protected static Properties getDebeziumProperties(JsonNode config, ConfiguredAir
     props.setProperty("value.converter.schemas.enable", "false");
 
     // https://debezium.io/documentation/reference/configuration/event-flattening.html
-    props.setProperty("delete.handling.mode", "rewrite");
+    // props.setProperty("delete.handling.mode", "rewrite");
```
Comment (cgardens) [inline airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/DebeziumRecordPublisher.java:126]: work in progress on this property.

File: airbyte-commons/src/main/java/io/airbyte/commons/util/AutoCloseableIterators.java [hunk 2]
```diff
@@ -72,6 +72,12 @@
     return new DefaultAutoCloseableIterator<>(stream.iterator(), stream::close);
   }
 
+  public static <T> List<T> toList(AutoCloseableIterator<T> iterator) throws Exception {
```
Comment (michel-tricot) [inline airbyte-commons/src/main/java/io/airbyte/commons/util/AutoCloseableIterators.java:75]: Can you add a comment to expose that it closes the iterator?

File: airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/DebeziumRecordIterator.java [hunk 3]
```diff
@@ -148,7 +148,10 @@ private PgLsn extractLsn(ChangeEvent<String, String> event) {
         .flatMap(source -> Optional.ofNullable(source.get("lsn").asText()))
         .map(Long::parseLong)
         .map(PgLsn::fromLong)
-        .orElseThrow(() -> new IllegalStateException("Could not find LSN"));
+        .orElseThrow(() -> {
+          System.out.println("event = " + event);
```
Comment (michel-tricot) [inline airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/DebeziumRecordIterator.java:152]: debug statement

File: airbyte-integrations/connectors/source-appsflyer-singer/unit_tests/unit_test.py [hunk 4]
```diff
@@ -22,6 +22,5 @@
 SOFTWARE.
 """
 
-
 def test_example_method():
```
Comment (davinchia) [inline airbyte-integrations/connectors/source-appsflyer-singer/unit_tests/unit_test.py:25]: accidental?

File: airbyte-commons/src/main/java/io/airbyte/commons/util/AutoCloseableIterators.java (hunk 2 above)
Comment (davinchia) [reply airbyte-commons/src/main/java/io/airbyte/commons/util/AutoCloseableIterators.java:75]: might be clearer to name this function something like `toListAndClose`.

**Review by davinchia — APPROVED** [review]
Makes sense! Briefly looked through to general flow. I did not examine the test cases in detail as I assume you and Jared have the spectrum covered. Let me know if you wanted a specific focus on an area.

One thoughts on testing. What do you think about using the @DisplayName annotation with Gherkin verbs (Given, When, Should etc) to better summarise what a test is doing? e.g. with the `testWhitelistFiltersFullRefresh` the display name can read: "When reading multiple streams should only return whitelist for incremental streams". Usage is up to dev discretion. Although there is a small cost, I find it really saves time with understanding tests, especially for beefier tests. e.g. `testRecordsProducedDuringAndAfterSync`, since the reader doesn't need to parse everything in the test block. Just a thought; not blocking.

File: airbyte-integrations/connectors/source-appsflyer-singer/unit_tests/unit_test.py (hunk 4 above)
Comment (cgardens) [reply airbyte-integrations/connectors/source-appsflyer-singer/unit_tests/unit_test.py:25]: i think these are the actual correct style? will make sure to run style check again.

File: airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/DebeziumRecordIterator.java (hunk 3 above)
Comment (cgardens) [reply airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/DebeziumRecordIterator.java:152]: thanks! still debugging this bit with the delete issue i mentioned.

File: airbyte-commons/src/main/java/io/airbyte/commons/util/AutoCloseableIterators.java (hunk 2 above)
Comment (cgardens) [reply airbyte-commons/src/main/java/io/airbyte/commons/util/AutoCloseableIterators.java:75]: yup

**Review by jrhizor — APPROVED** [review]

**Pull request merged**